### PR TITLE
made hlsmonitor more restful

### PR DESCRIPTION
**Pull request by Sgtfishtank**
Fixes: #8

### CODE REVIEW

File: src/HLSMonitorService.ts [hunk 1]
```diff
@@ -47,42 +45,115 @@ export class HLSMonitorService {
...
+        monitor = new HLSMonitor(body["streams"], body["stale_limit"]);
+      } else {
+        monitor = new HLSMonitor(body["streams"]);
+      }
+      monitor.create();
+      this.hlsMonitors.set(uuid,monitor)
+      console.log(this.monitors)
+      const rep = {
+        status: "Created a new hls-monitor",
+        streams: body["streams"]
```
Comment (oscnord) [inline src/HLSMonitorService.ts:94]: Just a question but why do you generate the monitor id here and not just in in the `HLSMonitor` constructor when instantiating a new one?

File: src/HLSMonitorService.ts [hunk 2]
```diff
@@ -47,42 +45,115 @@ export class HLSMonitorService {
...
+      const logs = await this.hlsMonitors.get(request.params.monitorId).getErrors()
       reply
         .code(200)
         .header('Content-Type', 'application/json; charset=utf-8')
         .send({ logs: logs });
     });
 
-    this.fastify.get("/clear-errors", async (request, reply) => {
-      if (!this.hlsMonitor) {
+    this.fastify.get("/monitor/:monitorId/clear-errors", async (request, reply) => {
```
Comment (birme) [inline src/HLSMonitorService.ts:63]: Perhaps we should have a DELETE on `/monitor/:monitorId/status` instead? I think that would be more restful

**Review by birme — CHANGES_REQUESTED** [review]
Looks overall good. Just a small suggestion to use another endpoint and method to clear errors.

File: src/HLSMonitorService.ts (hunk 1 above)
Comment (birme) [reply src/HLSMonitorService.ts:94]: I think that is fine considering that it is the HLSMonitorService that manages a set of monitors.

File: src/HLSMonitorService.ts (hunk 2 above)
Comment (Sgtfishtank) [reply src/HLSMonitorService.ts]: I changed it to DELETE, not sure it is the correct method but lets go with it for now

File: src/HLSMonitorService.ts [hunk 3]
```diff
@@ -92,43 +166,43 @@ export class HLSMonitorService {
...
+      this.hlsMonitors.get(request.params.monitorId).start();
       reply
         .code(200)
         .header('Content-Type', 'application/json; charset=utf-8')
         .send({ status: "Started monitoring" });
     });
 
-    this.fastify.put("/delete", async (request, reply) => {
-      if (!this.hlsMonitor) {
+    this.fastify.put("/monitor/:monitorId/delete", async (request, reply) => {
```
Comment (birme) [inline src/HLSMonitorService.ts:197]: I suggest using HTTP method DELETE on this resource instead, i.e.

`HTTP DELETE /monitor/:monitorId`

File: src/HLSMonitorService.ts [hunk 4]
```diff
@@ -47,42 +45,118 @@ export class HLSMonitorService {
...
+      const logs = await this.hlsMonitors.get(request.params.monitorId).getErrors()
       reply
         .code(200)
         .header('Content-Type', 'application/json; charset=utf-8')
         .send({ logs: logs });
     });
 
-    this.fastify.get("/clear-errors", async (request, reply) => {
-      if (!this.hlsMonitor) {
+    this.fastify.delete("/monitor/:monitorId/clear-errors", async (request, reply) => {
```
Comment (birme) [inline src/HLSMonitorService.ts:63]: I think it would be more restful to have:

`HTTP DELETE /monitor/:monitorId/status`

to clear errors.

**Review by birme — CHANGES_REQUESTED** [review]
Some changes to API suggested

File: src/HLSMonitorService.ts (hunk 2 above)
Comment (birme) [reply src/HLSMonitorService.ts]: Yes, but change the resource to `/monitor/:monitorId/status` instead of `clear-errors`

**Review by birme — APPROVED** [review]

**Pull request merged**